### ISSUE TITLE
GridSearch and results

### ISSUE THREAD

**Issue by Garfield2013** [catboost/catboost#1084]
```
model.fit(train_data, y=label_data, eval_set=eval_dataset)
eval_dataset = Pool(val_data, val_labels)
model = CatBoostClassifier(depth=8 or 10, iterations=10, task_type="GPU", devices='0-2', eval_metric='Accuracy', boosting_type="Ordered", bagging_temperature=0, use_best_model=True)
```

When I run the code above (in 2 separate runs) I get the following results:

Depth 10: 0.6864865
Depth 8:   0.6756757

I would like to setup and run GridSearch in a way - so it runs exactly the same combinations and produces the exact same results - as when I run the code manually.

GridSearch code:

```
model = CatBoostClassifier(iterations=10, task_type="GPU", devices='0-2', eval_metric='Accuracy', boosting_type="Ordered", depth=10, bagging_temperature=0, use_best_model=True)

grid = {'depth': [8,10]}
grid_search_result = GridSearchCV(model, grid, cv=2)
results = grid_search_result.fit(train_data, y=label_data, eval_set=eval_dataset) 
```

Issues: 

1. I would like the GridSearch to use my "eval_set" to compare/validate all the different runs (like when run manually) - But it uses something else which I don't understand what is and it doesn't seems to look at "eval_set" at all?

2. It produces not only 2 results - but depending on the "cv" (The cross-validation splitting strategy.) param it run 3,5,7,9 or 11 runs? I don't want that.

3. I tried to go through the entire "results" object via the debugger - but I simply can't find the validation "Accuracy" scores for the best or all the other runs. I can find a lot of other values - but none of them matches what I'm looking for. The numbers don't match the numbers the "eval_set" dataset produces?

**Comment by annaveronika** [comment]
I think your issue is about GridSearchCV, which is a part of sklearn, not a part of catboost.

**Comment by Garfield2013** [comment]
But can you help me to understand - so I can make the GridSearch produces the exact same results as CATBoost when run manually?

**Comment by annaveronika** [comment]
There is a set of places, where you will get help with answering questions about different ml libraries. The biggest one is stackoverflow, where you'll definitely get help from sklearn users, or where you will in many cases find a ready answer.
Other places for catboost related discussions are catboost telegram chats and #tool_catboost in ods slack channel.

One thing about exact same results with CatBoost: CatBoost GPU is not deterministic, because of different order of floating number summations, so with GPU you will always get slightly different results.

**Comment by Garfield2013** [comment]
Thank you (I know :) ).
I can reproduce the same result on GPU...The validation accuracy and the best iteration is always the same when the same settings are in use... that's not the problem... :) I just thought that it was a common challenge in regards to Catboost (thats why I ask you)... How do you prefer to run many runs - so they match manually runs via CATBoost?

**Comment by Garfield2013** [comment]
I created it as and SO post as well (I really like to hear your answer as well Anna :) ):

https://stackoverflow.com/questions/58951164/catboost-and-gridsearch

**Comment by annaveronika** [comment]
Thanks! Closing the issue then.
